### PR TITLE
Upgrade submodule oneDNN to v3.5.3

### PR DESCRIPTION
**Pull request by yanbing-j**
This PR is to upgrad submodule oneDNN to v3.5.3.

## Improvements

- [experimental] Introduced [microkernel API](https://oneapi-src.github.io/oneDNN/ukernels.html) for Intel Architecture Processors. This API exposes internal mechanisms used in matmul and convolution implementation to expert users.
- Improved performance of matmul primitive with sum post-op for batched cases on processors with Intel AMX instruction set support.
- Introduced fp64 matmul support. This functionality is currently implemented on Intel GPUs with hardware acceleration for fp64 math only.

## Validation results on CPU
No regression was found.

1. NLP models accuracy/inference/training

Model Name | Mode Name | Precision | OneDNN | Baseline | OneDNN/Baseline
-- | -- | -- | -- | -- | --
bert-large | realtime | bf16 | 192.498 | 189.664 | 1.014942214
bert-large | throughput | bf16 | 202.424 | 202.156 | 1.001325709
bert-large | train_phase2 | bf16 | 15.955 | 16.029 | 0.995383368
LCM | throughput | bf16 | 1.01983 | 1.06632 | 0.956401455
stable-diffusion | throughput | bf16 | 0.10313 | 0.10184 | 1.012666929
ViT | realtime | bf16 | 1086.48 | 928.43 | 1.17023362
ViT | throughput | bf16 | 1419.07 | 1393.81 | 1.018122987
yolov7 | realtime | bf16 | 413.468682 | 415.16503 | 0.995914039
yolov7 | throughput | bf16 | 369.697 | 366.789 | 1.007928264
bert-large | realtime | fp32 | 46.685 | 46.652 | 1.000707365
bert-large | throughput | fp32 | 47.766 | 48.007 | 0.994979899
bert-large | train_phase2 | fp32 | 7.101 | 7.104 | 0.999577703
LCM | throughput | fp32 | 0.5501 | 0.55023 | 0.999763735
stable-diffusion | throughput | fp32 | 0.04012 | 0.04002 | 1.002498751
ViT | realtime | fp32 | 337.27 | 335.19 | 1.006205436
ViT | throughput | fp32 | 346.52 | 350.08 | 0.989830896
yolov7 | realtime | fp32 | 107.138054 | 107.242747 | 0.999023775
yolov7 | throughput | fp32 | 103.383 | 104.301 | 0.99119855
bert-large | realtime | int8 | 283.541 | 289.569 | 0.979182855
LCM | throughput | int8 | 1.09864 | 1.08998 | 1.0079451
stable-diffusion | throughput | int8 | 0.10617 | 0.10604 | 1.001225952
ViT | realtime | int8 | 1562.11 | 1554.68 | 1.004779119
ViT | throughput | int8 | 1904.38 | 1903.39 | 1.000520125
yolov7 | realtime | int8 | 540.489493 | 539.902488 | 1.001087243
yolov7 | throughput | int8 | 499.999 | 500.757 | 0.998486292

Device | Dtype | Geomean Higher is better
-- | -- | --
All | all | 101.17%
All | fp32 | 99.83%
All | bf16 | 102.24%
All | int8 | 99.91%
All | fp16 | 103.61%
SPR | all | 100.54%
SPR | fp32 | 99.82%
SPR |bf16 | 101.78%
SPR |int8 | 99.90%
GNR | all | 101.58%
GNR | fp32 | 99.85%
GNR | bf16 | 102.66%
GNR | int8 | 99.93%
GNR | fp16 | 103.61%

2. Torchbench cpu userbenchmark inference & training

Perf_Geomean | Ratio (oneDNN/baseline)
-- | --
eager_throughtput_bf16_infer | 1.00x
eager_throughtput_fp32_infer | 1.00x
jit_llga_throughtput_amp_bf16 | 1.00x
jit_llga_throughtput_fp32 | 1.00x
eager_throughtput_fx_int8 | 0.99x
eager_throughtput_bf16_train | 1.01x
eager_throughtput_fp32_train | 1.00x

3. Inductor quantization

Static quant:
Perf_Geomean | Ratio (oneDNN/baseline)
-- | --
PTQ | 1.00x
PTQ_CPP_WRAPPER | 1.00x
QAT | 1.00x

ACC_Geomean | Ratio (oneDNN/baseline)
-- | --
PTQ | 1.00x
PTQ_CPP_WRAPPER | 1.00x
QAT | 1.00x

Dynamic quant:

  | Ratio (oneDNN/baseline)
-- | --
Performance | 1.04x
Accuracy | 1.00x

4. Dynamo benchmarks
GEOMEAN summary
image

FP32 Static shape, default wrapper
image

FP32 Dynamic shape, default wrapper
image

AMP Static shape, default wrapper
image

AMP Dynamic shape, default wrapper
image

## Validation results on XPU
Category | Eager | Inductor
-- | -- | --
huggingface_amp_fp16_training | 1.002456 | 0.999998
huggingface_bfloat16_inference | 1.005386 | 1.003511
huggingface_float32_training | 1.002533 | 1.003098
torchbench_amp_fp16_training | 1.009065 | 1.01323
torchbench_bfloat16_inference | 1.003371 | 1.001534
torchbench_float32_training | 1.012102 | 1.011596
timm_models_amp_fp16_training | 1.005511 | 1.010329
timm_models_bfloat16_inference | 1.000935 | 1.000538
timm_models_float32_training | 0.991873 | 0.99721

cc @gujinghui @PenghuiCheng @XiaobingSuper @jianyuh @jgong5 @mingfeima @sanchitintel @ashokei @jingxu10 @min-jean-cho @Guobing-Chen @Xia-Weiwen @snadampal

### CODE REVIEW

**Comment by Xia-Weiwen** [comment]
Previous discussions are here: https://github.com/pytorch/pytorch/pull/129932

**Comment by yanbing-j** [comment]
https://github.com/pytorch/pytorch/pull/130618 can fix aarch64 CI failures.

**Comment by murste01** [comment]
Hi All, these are our validation results for AArch64:

**Torchbench (FP32 Only):**

| Test Suite                       | Ratio Geomean (New/Baseline) |
|----------------------------------|------------------------------|
| eager_throughput_fp32_infer      | 1.41                  |
| eager_throughput_fp32_train      | 1.26                  |
| jit_llga_throughput_fp32         | 1.09                  |
| eager_throughput_fx_int8         | 1.07                   |

**Dynamo (FP32 Only, Eager):**

| Precision | Shape   | Wrapper | Threads | Ratio Geomean (Baseline/New) |
|-----------|---------|---------|---------|------------------------------|
| fp32      | Static  | Default | 1       | 1.05                  |
| fp32      | Static  | Default | 16      | 1.23                  |
| fp32      | Dynamic | Default | 1       | 1.05                  |
| fp32      | Dynamic | Default | 16      | 1.24                  |

**Dynamo (FP32 Only, Inductor):**

| Precision | Shape   | Wrapper | Threads | Ratio Geomean (Baseline/New) |
|-----------|---------|---------|---------|------------------------------|
| fp32      | Static  | Default | 1       | 1.03                 |
| fp32      | Static  | Default | 16      | 1.19                  |
| fp32      | Dynamic | Default | 1       | 1.03                  |
| fp32      | Dynamic | Default | 16      | 1.21                  |

**Review by jgong5 — APPROVED** [review]
Need to wait for the fix for CI failures before landing.

**Comment by sanchitintel** [comment]
Hi @yanbing-j, are the CI failures legit?

**Comment by yanbing-j** [comment]
@sanchitintel Still waiting for the fix of CI failures.

**Comment by Guobing-Chen** [comment]
@murste01 and @milpuz01, thanks for your validation/verification on tag v3.5.1. As we are moving to v3.5.3, could you have a check about the git commit difference (shared with you in email thread already) and see whether additional testing needed?

**Comment by yanbing-j** [comment]
@malfet @atalman Could you please help review this PR?

For oneDNN v3.5.1, we have finished CPU, XPU and ARM tests with no regression. And for oneDNN v3.5.3, CPU and XPU tests are refreshed. It still needs ARM folk's conformation (whether more validation on v3.5.3 needed, based on their v3.5.1 test result).

**Comment by sanchitintel** [comment]
> @murste01 and @milpuz01, thanks for your validation/verification on tag v3.5.1. As we are moving to v3.5.3, could you have a check about the git commit difference (shared with you in email thread already) and see whether additional testing needed?

Hi @murste01 and @milpuz01, any updates? Thanks!

**Review by malfet — APPROVED** [review]
Looks good to me, provided there are no regressions on ARM end

**Comment by yanbing-j** [comment]
Hi @murste01 and @milpuz01, this PR is merged without PyTorch CI failures and internal test failres. However, we still do not receive ARM test results based on oneDNN v3.5.3. Could you please post ARM test results when available? Thanks!
cc @Guobing-Chen

**Comment by murste01** [comment]
Hi @yanbing-j. I have the results now.

**Torchbench (FP32 Only):**

| Test Suite                  | Ratio Geomean (New/Baseline) |
|-----------------------------|------------------------------|
| eager_throughput_fp32_infer | 1.02                  |
| eager_throughput_fp32_train | 1.04                  |
| jit_llga_throughput_fp32    | 0.97                  |
| eager_throughput_fx_int8    | 1.01                  |

**Dynamo (FP32 Only, Eager):**

| Precision | Shape   | Wrapper | Threads | Ratio Geomean (Baseline/New) |
|-----------|---------|---------|---------|------------------------------|
| fp32      | Dynamic | default | 16      | 1.12                  |
| fp32      | Dynamic | default | 1       | 1.14                  |
| fp32      | Static  | default | 16      | 1.12                  |
| fp32      | static  | default | 1       | 1.15                  |

**Dynamo (FP32 Only, Inductor):**

| Precision | Shape   | Wrapper | Threads | Ratio Geomean (Baseline/New) |
|-----------|---------|---------|---------|------------------------------|
| fp32      | Dynamic | default | 16      | 1.08                  |
| fp32      | Dynamic | default | 1       | 1.14                  |
| fp32      | Static  | default | 16      | 1.09                  |
| fp32      | Static  | default | 1       | 1.13                  |

Overall improvements, but I'll be having a look at our jit_llga_throughput_fp32 results.

**Comment by yanbing-j** [comment]
@murste01 Thanks for the followup!
